src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -26,23 +26,19 @@ public AuthenticationHandler(bool preAuthenticate, ICredentials credentials, Htt
             _innerHandler = innerHandler;
         }
 
-        private bool TrySetBasicAuthToken(HttpRequestMessage request)
-        {
-            NetworkCredential credential = _credentials.GetCredential(request.RequestUri, Basic);
-            if (credential == null)
-            {
-                return false;
-            }
-
-            request.Headers.Authorization = new AuthenticationHeaderValue(Basic, BasicAuthenticationHelper.GetBasicTokenForCredential(credential));
-            return true;
-        }
-
         protected internal override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
             if (_preAuthenticate)
             {
-                TrySetBasicAuthToken(request);
+                // Try using previous digest response WWWAuthenticate header
+                if (_digestResponse != null)
+                {
+                    AuthenticationHelper.TrySetDigestAuthToken(request, _credentials, _digestResponse);
+                }
+                else
+                {
+                    AuthenticationHelper.TrySetBasicAuthToken(request, _credentials);
```

Comment:
I'm not sure how we can guard against that here, I assume preauthenticate will be used in https scenario, where the connection will be a ssl stream. The preauthentication property provides facility to try authentication before a challenge from server. So it's obvious that digest is not possible for first request, in which case the user has to be aware and make a conscious decision to set preauthenticate or not for a request.